plugins/outputs/postgresql/postgresql_test.go

```diff
@@ -0,0 +1,623 @@
+}
+
+func (la *LogAccumulator) Warnf(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelWarn, format, args)
+}
+
+func (la *LogAccumulator) Warn(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelWarn, "%v", args)
+}
+
+func (la *LogAccumulator) Infof(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelInfo, format, args)
+}
+
+func (la *LogAccumulator) Info(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelInfo, "%v", args)
+}
+
+var ctx = context.Background()
+
+func TestMain(m *testing.M) {
+	// Try and find the server.
+	// Try provided env vars & defaults first.
+	if c, err := pgx.Connect(ctx, ""); err != nil {
+		os.Setenv("PGHOST", "127.0.0.1")
+		os.Setenv("PGUSER", "postgres")
+		// Try the port used in docker-compose.yml first
+		os.Setenv("PGPORT", "5433")
+		if c, err := pgx.Connect(ctx, ""); err != nil {
+			// Fall back to the default port
+			os.Setenv("PGPORT", "5432")
+		} else {
+			c.Close(ctx)
+		}
+	} else {
+		c.Close(ctx)
+	}
+
+	if err := prepareDatabase("telegraf"); err != nil {
+		fmt.Fprintf(os.Stderr, "Error preparing database: %s\n", err)
+		os.Exit(1)
+	}
+	os.Exit(m.Run())
+}
+
+func prepareDatabase(name string) error {
+	db, err := pgx.Connect(ctx, "")
+	if err != nil {
+		return err
+	}
+	_, err = db.Exec(ctx, "DROP DATABASE IF EXISTS "+name)
+	if err != nil {
+		return err
+	}
+	_, err = db.Exec(ctx, "CREATE DATABASE "+name)
+	return err
+}
+
+type PostgresqlTest struct {
+	Postgresql
+	Logger *LogAccumulator
+}
+
+func newPostgresqlTest(tb testing.TB) *PostgresqlTest {
+	p := newPostgresql()
+	logger := NewLogAccumulator(tb)
+	p.Logger = logger
+	pt := &PostgresqlTest{Postgresql: *p}
+	pt.Logger = logger
+	pt.Connection = "database=telegraf"
+	pt.LogLevel = "debug"
+	return pt
+}
+
+func TestPostgresqlConnect(t *testing.T) {
+	p := newPostgresqlTest(t)
+	require.NoError(t, p.Connect())
+	assert.EqualValues(t, 1, p.db.Stat().MaxConns())
+	p.Close()
+
+	p = newPostgresqlTest(t)
+	p.Connection += " pool_max_conns=2"
+	require.NoError(t, p.Connect())
+	assert.EqualValues(t, 2, p.db.Stat().MaxConns())
+	p.Close()
+}
+
+func newMetric(
+	t *testing.T,
+	suffix string,
+	tags map[string]string,
+	fields map[string]interface{},
+) telegraf.Metric {
+	return metric.New(t.Name()+suffix, tags, fields, time.Now())
+}
+
+type MSS = map[string]string
+type MSI = map[string]interface{}
+
+func dbTableDump(t *testing.T, db *pgxpool.Pool, suffix string) []MSI {
+	rows, err := db.Query(ctx, "SELECT * FROM "+pgx.Identifier{t.Name() + suffix}.Sanitize())
+	require.NoError(t, err)
+	defer rows.Close()
+
+	var dump []MSI
+	for rows.Next() {
+		msi := MSI{}
+		vals, err := rows.Values()
+		require.NoError(t, err)
+		for i, fd := range rows.FieldDescriptions() {
+			msi[string(fd.Name)] = vals[i]
+		}
+		dump = append(dump, msi)
+	}
+	require.NoError(t, rows.Err())
+	return dump
+}
+
+func TestWrite_sequential(t *testing.T) {
```

Comment:
Please remove the underscores in the test names and use CamelCase.

Reply:
That's not idomatic go. Convention is to have the thing being tested in camel case, and the the behavior you're testing as a suffix with undescore. Golang's on documentation covers this: https://golang.org/pkg/testing/